packages/squiggle-lang/src/rescript/ReducerInterface/ReducerInterface_GenericDistribution.res

```diff
@@ -0,0 +1,163 @@
+    | "beta" => Ok(SymbolicDist.Beta.make)
+    | "lognormal" => Ok(SymbolicDist.Lognormal.make)
+    | "to" => Ok(SymbolicDist.From90thPercentile.make)
+    | _ => Error("impossible path")
+    }
+
+  let threeFloat = name =>
+    switch name {
+    | "triangular" => Ok(SymbolicDist.Triangular.make)
+    | _ => Error("impossible path")
+    }
+
+  let symbolicResultToOutput = (
+    symbolicResult: result<SymbolicDistTypes.symbolicDist, string>,
+  ): option<GenericDist_GenericOperation.outputType> =>
+    switch symbolicResult {
+    | Ok(r) => Some(Dist(Symbolic(r)))
+    | Error(r) => Some(GenDistError(Other(r)))
+    }
+}
+
+let dispatchToGenericOutput = (call: ExpressionValue.functionCall): option<
+  GenericDist_GenericOperation.outputType,
+> => {
+  let (fnName, args) = call
+  switch (fnName, args) {
+  | ("exponential" as fnName, [EvNumber(f1)]) =>
+    SymbolicConstructors.oneFloat(fnName)
+    ->E.R.bind(r => r(f1))
+    ->SymbolicConstructors.symbolicResultToOutput
+  | (
+      ("normal" | "uniform" | "beta" | "lognormal" | "to") as fnName,
+      [EvNumber(f1), EvNumber(f2)],
+    ) =>
+    SymbolicConstructors.twoFloat(fnName)
+    ->E.R.bind(r => r(f1, f2))
+    ->SymbolicConstructors.symbolicResultToOutput
+  | ("triangular" as fnName, [EvNumber(f1), EvNumber(f2), EvNumber(f3)]) =>
+    SymbolicConstructors.threeFloat(fnName)
+    ->E.R.bind(r => r(f1, f2, f3))
+    ->SymbolicConstructors.symbolicResultToOutput
+  | ("sample", [EvDistribution(dist)]) => Helpers.toFloatFn(#Sample, dist)
+  | ("mean", [EvDistribution(dist)]) => Helpers.toFloatFn(#Mean, dist)
+  | ("normalize", [EvDistribution(dist)]) => Helpers.toDistFn(Normalize, dist)
+  | ("toPointSet", [EvDistribution(dist)]) => Helpers.toDistFn(ToPointSet, dist)
+  | ("cdf", [EvDistribution(dist), EvNumber(float)]) => Helpers.toFloatFn(#Cdf(float), dist)
+  | ("pdf", [EvDistribution(dist), EvNumber(float)]) => Helpers.toFloatFn(#Pdf(float), dist)
+  | ("inv", [EvDistribution(dist), EvNumber(float)]) => Helpers.toFloatFn(#Inv(float), dist)
+  | ("toSampleSet", [EvDistribution(dist), EvNumber(float)]) =>
+    Helpers.toDistFn(ToSampleSet(Belt.Int.fromFloat(float)), dist)
+  | ("truncateLeft", [EvDistribution(dist), EvNumber(float)]) =>
+    Helpers.toDistFn(Truncate(Some(float), None), dist)
+  | ("truncateRight", [EvDistribution(dist), EvNumber(float)]) =>
+    Helpers.toDistFn(Truncate(None, Some(float)), dist)
+  | ("truncate", [EvDistribution(dist), EvNumber(float1), EvNumber(float2)]) =>
+    Helpers.toDistFn(Truncate(Some(float1), Some(float2)), dist)
+  | (("add" | "multiply" | "subtract" | "divide" | "exponentiate" | "log") as arithmetic, [a, b] as args) =>
+    Helpers.catchAndConvertTwoArgsToDists(args)->E.O2.fmap(((fst, snd)) =>
+      Helpers.twoDiststoDistFn(Algebraic, arithmetic, fst, snd)
+    )
+  | (
+      ("dotAdd" | "dotMultiply" | "dotSubtract" | "dotDivide" | "dotExponentiate" | "dotLogarithm") as arithmetic,
```

Comment:
we've been calling these `pointwise` mostly haven't we? the `dot` refers to the syntax but we may prefer to make the string follow the AST node

Reply:
Good point! I like that more.